network/src/protocols/wire/handshake/v1/mod.rs

```diff
@@ -113,15 +113,20 @@ impl fmt::Display for ProtocolId {
 }
 
 //
-// SupportedProtocols
+// ProtocolIdSet
 //
 
-/// A bit vector of supported [`ProtocolId`]s.
+/// A compact representation for a set of [`ProtocolId`]s. Internally, this is a
+/// bitvec which supports at most 256 bits.
+///
+/// These sets are sent over-the-wire in the initial [`HandshakeMsg`] to other
+/// DiemNet peers in order to negotiate the set of common supported protocols for
+/// use on a new DiemNet connection.
```

Comment:
So, why did we end up making such a compact format for protocols that doesn't leave room for bit flips or protections against them?

I guess we should be fine, just seems like it was an overoptimization for a handshake that doesn't happen all so often.

Reply:
i'll be honest, I don't really remember why haha 😂 

at the end of the day, it's really just a `Set<ProtocolId>`. it could be represented as a `Vec<ProtocolId>`. you could also do some more fancy thing, but this works for now. ideally we almost never need to change the handshake. if we do, we have a process (new nodes advertise a new handshake in their network address, i.e., `/ln-handshake/1` vs `/ln-handshake/0`, then nodes old nodes will dial the address with the old handshake, and new nodes support both for a while).